### PR TITLE
UIIN-2041 Select existing order when creating an order from instance …

### PR DESCRIPTION
**Pull request by usavkov-epam**
…record

## Purpose

https://issues.folio.org/browse/UIIN-2041
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/ViewInstance.js [hunk 1]
```diff
@@ -54,6 +55,7 @@ import { CalloutRenderer } from './components';
 
 import ImportRecordModal from './components/ImportRecordModal';
 import NewInstanceRequestButton from './components/ViewInstance/MenuSection/NewInstanceRequestButton';
+import NewOrderModal from './components/NewOrderModal';
```
Comment (AndreiShumski2) [inline src/ViewInstance.js:58]: just `'./components'`, please export NewOrderModal from `components/index.js`

File: src/ViewInstance.js [hunk 2]
```diff
@@ -139,6 +141,13 @@ class ViewInstance extends React.Component {
         query: '(module==SETTINGS and configName==TLR)',
       },
     },
+    orders: {
```
Comment (AndreiShumski2) [inline src/ViewInstance.js:144]: Is it required? Maybe, make `ky.get` request for order number validation in `NewOrderModalContainer` component instead?

File: src/ViewInstance.js [hunk 3]
```diff
@@ -808,6 +821,17 @@ class ViewInstance extends React.Component {
               />
             </IfPermission>
           </IfInterface>
+
+          <IfInterface name="orders">
+            <IfPermission perm="ui-inventory.instance.createOrder">
```
Comment (AndreiShumski2) [inline src/ViewInstance.js:826]: IMO it's redundant to check interface and permissions for modal. It's already checking for rendering action button that call this Modal.

**Review by AndreiShumski2 — COMMENTED** [review]
Looks good apart some small comments

File: src/components/NewOrderModal/NewOrderModal.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      <Button
+        marginBottom0
+        onClick={onCancel}
+      >
+        <FormattedMessage id="ui-inventory.cancel" />
+      </Button>
+    </ModalFooter>
+  ));
+
+  const message = useMemo(() => (
```
[the remark this answers is not in the document]
Comment (usavkov-epam) [reply src/components/NewOrderModal/NewOrderModal.js]: Definitely not, good point, I missed it)) Thx!

File: src/ViewInstance.js (hunk 1 above)
Comment (usavkov-epam) [reply src/ViewInstance.js]: Done

File: src/ViewInstance.js (hunk 3 above)
Comment (usavkov-epam) [reply src/ViewInstance.js]: I reinsured))
Removed now

**Review by AndreiShumski2 — APPROVED** [review]

**Review by mkuklis — APPROVED** [review]

**Pull request merged**